[8.1] Parsing the jdl input by using the Job API (#6300)
DIRACGrid/DIRAC: docs/source/DeveloperGuide/AddingNewComponents/DevelopingInterfaces/index.rst

            return super().setInputData(lfns.append("LFN:/some/file.lfn"))

        def setCustomElement(self, element: str):
            self._addJDLParameter("CustomElement", element)

Note that each method, class and module should have its own docstring. This is important for the documentation to be generated correctly.

In our example, our newly created CustomJob inherits from the Job class. This means that it will have all the methods and attributes of Job, and that we can add new ones to it.

The first method is the constructor, i.e the ``__init__`` method. The first call is the constructor of the super Job class, in which we're able to set the standard output and error files to "custom.out" and "custom.err" respectively. Then, we also set the name of the job to "CustomJob" by calling one of the Job method. This allow to avoid code duplication and any breakage in case of a change of implementation to the setName method implementation.

The second method ``setInputData`` shows that VOs can override some methods of the Job class, here by adding some file when setting the input data. This is done by calling the super method and appending the new LFN to the list of LFNs.

> Finally, the third method ``setCustomElement`` shows that VOs can add new methods to the Job class. Here, we add a new JDL parameter called "CustomElement" and set it to the value of the element parameter. This is done by calling the ``_addJDLParameter`` method is discouraged. However, if you really need to add some custom JDL parameters, more steps may be needed.

Reviewer: Do you mean that using `_addJDLParameter()` is not recommended?
 
```suggestion
Finally, the third method ``setCustomElement`` shows that VOs can add new methods to the Job class. Here, we add a new JDL parameter called "CustomElement" and set it to the value of the element parameter. This is done by calling the ``_addJDLParameter`` method, which is not recommended. If you really need to add some custom JDL parameters, more steps may be needed.
```

In DIRAC, a user has two ways to create jobs.

The first one is by using the Job API :

.. figure:: submitting-using-job-api.png
    :alt: Submission process using the Job API
    :align: center

The second one is by using the JDL format, documented here :doc:`../../../UserGuide/Tutorials/JDLsAndJobManagementBasic/index`

.. figure:: submitting-using-jdl-directly.png
    :alt: Submission process using the JDL format directly
    :align: center

> However, as you may see, the WMSClient class is not called with the same arguments in both cases. This is because the Job API class is also responsable for creating a Workflow file. To unify the two submission methods, a ``loadJDLAsJob`` has been added. The sequence diagram below shows how the Dirac API uses this new method to transform JDL jobs into jobs with a workflow script.

Reviewer: ```suggestion
However, as you may see, the WMSClient class is not called with the same arguments in both cases. This is because the Job API class is also responsible for creating a Workflow file. To unify the two submission methods, a ``loadJDLAsJob`` has been added. The sequence diagram below shows how the Dirac API uses this new method to transform JDL jobs into jobs with a workflow script.
```
